addons/hr_expense/models/hr_expense.py

```diff
@@ -406,7 +406,7 @@ def _compute_price_unit(self):
             if expense.state not in {'draft', 'reported'}:
                 continue
             product_id = expense.product_id
-            if product_id and expense.product_has_cost and not expense.nb_attachment:
+            if product_id and expense.product_has_cost:
```

Comment:
Instead of overriding the entire compute, why not simply create a method that will check whether the `price_unit` should be computed using the product's `_price_compute`.
You can then override that method in `he_expense_extract`.